src/models/stores/serial.ts

```diff
@@ -2,16 +2,21 @@ import { NodeChannelInfo } from "src/plugins/dataflow-tool/model/utilities/node"
 
 export class SerialDevice {
   value: string;
+  outValue: number;
   localBuffer: string;
   private port: SerialPort | null;
   connectChangeStamp: number | null;
   lastConnectMessage: string | null;
   deviceInfo: SerialPortInfo | null;
   serialNodesCount: number;
-  //localPorts: any;
+  writer: WritableStreamDefaultWriter;
+  userSeenSerialModal: boolean | null;
```

Comment:
Perhaps `serialModalShown`? Seems easier to parse (for me anyway) and has the advantage that it's stated in terms of the code's actions (the modal has been shown) as opposed to the user's behavior, which we can't _really_ know -- what if the user is blind?

Reply:
Thanks, excellent point.